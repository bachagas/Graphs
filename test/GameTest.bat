java -Xms2g -Xss2g -cp "../bin;../lib/*" test.GameTest > GameTest.out