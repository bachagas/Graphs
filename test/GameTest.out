************************* Start **************************
Tue Nov 12 15:38:19 BRST 2013 (1384277899108)> Start
Testing for time measurement...Done!
Tue Nov 12 15:38:20 BRST 2013 (1384277900492)> Finished!
1384 milliseconds ellapsed
Tue Nov 12 15:38:20 BRST 2013 (1384277900492)> Testing permutations
All possible permutations of "0123" are:
Tue Nov 12 15:38:20 BRST 2013 (1384277900493)> Finished!
1 milliseconds ellapsed
[0123, 0132, 0213, 0231, 0312, 0321, 1023, 1032, 1203, 1230, 1302, 1320, 2013, 2031, 2103, 2130, 2301, 2310, 3012, 3021, 3102, 3120, 3201, 3210]

******************* Building Game Graph ******************
Tue Nov 12 15:38:20 BRST 2013 (1384277900494)> Building graph:
Must have 9!=362880 permutations, generated 362880 board configurations
Building a game graph with them all... Done!
Tue Nov 12 15:38:22 BRST 2013 (1384277902620)> Finished!
2126 milliseconds ellapsed
Game={362880 node(s),483840 edge(s)}

************************ Tarefa 1 ************************
Tue Nov 12 15:38:22 BRST 2013 (1384277902620)> Looking for connected components by DFS:
Performing DFS... Done!
Tue Nov 12 15:38:23 BRST 2013 (1384277903978)> Finished!
1358 milliseconds ellapsed
Found 2 connected components:
CC0={181440 node(s),241920 edge(s)}
CC1={181440 node(s),241920 edge(s)}

************************ Tarefa 2 ************************
Tue Nov 12 15:38:23 BRST 2013 (1384277903978)> Looking for maximum distance to solution by BFS:
Solution is in Connected Component CC0={181440 node(s),241920 edge(s)}
Performing BFS... Done!
Tue Nov 12 15:38:24 BRST 2013 (1384277904282)> Finished!
304 milliseconds ellapsed
Most far away node is:
Node label is "647850321"
Board is: 
[6 4 7 
 8 5 0 
 3 2 1]
Empty square is at position (1,2)
and it is at distance 31 to solution.
Path from it to solution is the following steps:
[6 4 7 
 8 5 0 
 3 2 1]
[6 4 7 
 8 0 5 
 3 2 1]
[6 0 7 
 8 4 5 
 3 2 1]
[6 7 0 
 8 4 5 
 3 2 1]
[6 7 5 
 8 4 0 
 3 2 1]
[6 7 5 
 8 4 1 
 3 2 0]
[6 7 5 
 8 4 1 
 3 0 2]
[6 7 5 
 8 4 1 
 0 3 2]
[6 7 5 
 0 4 1 
 8 3 2]
[6 7 5 
 4 0 1 
 8 3 2]
[6 0 5 
 4 7 1 
 8 3 2]
[0 6 5 
 4 7 1 
 8 3 2]
[4 6 5 
 0 7 1 
 8 3 2]
[4 6 5 
 7 0 1 
 8 3 2]
[4 0 5 
 7 6 1 
 8 3 2]
[4 5 0 
 7 6 1 
 8 3 2]
[4 5 1 
 7 6 0 
 8 3 2]
[4 5 1 
 7 6 2 
 8 3 0]
[4 5 1 
 7 6 2 
 8 0 3]
[4 5 1 
 7 0 2 
 8 6 3]
[4 0 1 
 7 5 2 
 8 6 3]
[4 1 0 
 7 5 2 
 8 6 3]
[4 1 2 
 7 5 0 
 8 6 3]
[4 1 2 
 7 5 3 
 8 6 0]
[4 1 2 
 7 5 3 
 8 0 6]
[4 1 2 
 7 5 3 
 0 8 6]
[4 1 2 
 0 5 3 
 7 8 6]
[0 1 2 
 4 5 3 
 7 8 6]
[1 0 2 
 4 5 3 
 7 8 6]
[1 2 0 
 4 5 3 
 7 8 6]
[1 2 3 
 4 5 0 
 7 8 6]
[1 2 3 
 4 5 6 
 7 8 0]

